engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -217,13 +296,32 @@ public function get($guid, $type = '') {
 			return false;
 		}
 
-		// Check local cache first
-		$new_entity = _elgg_services()->entityCache->get($guid);
-		if ($new_entity) {
+		$guid = (int) $guid;
+
+		$memcache = _elgg_get_memcache('new_entity_cache');
+		
+		$entity = $this->entity_cache->get($guid);
+		if (!$entity) {
+			$entity = $memcache->load($guid);
+		}
+
+		if (!$entity instanceof \ElggEntity) {
+			$entity = false;
+		}
+
+		// Validate accessibility
+		if ($entity && !elgg_get_ignore_access() && !has_access_to_entity($entity)) {
```

Comment:
Unless we invalidate in-memory cache when ignored access is toggled, and user is logged in/out, and only do this for memcache. But to be honest, I am quite disillusioned with memcache for entities - doesn't really add much gain for increased complexity.